invoices/invoiceregistry.go

```diff
@@ -1806,6 +1806,12 @@ func (i *InvoiceRegistry) notifyHodlSubscribers(htlcResolution HtlcResolution) {
 func (i *InvoiceRegistry) hodlSubscribe(subscriber chan<- interface{},
 	circuitKey CircuitKey) {
 
+	// If the caller decides to not subscribe to the resolution, we can
+	// exit early.
+	if subscriber == nil {
```

Comment:
?

Reply:
updated the commit msg and put the check to a more relevant callsite - in short, we wanna do a quick lookup for the preimage without the async subscription.